### DISCUSSION TITLE
Update interval

### DISCUSSION THREAD

**Discussion by HighPriest** [Miskamyasa/obsidian-codestats#5] [Polls]
Hi.
I have just looked through your creation and first off, I would like to thank you for making this. It is very useful, maybe you should suggest for it to be added to official list on the codestats website?

What I would like to ponder on is the interval with which the updates are sent to the server, which is currently set to 10 seconds and only when some XP is waiting to be uploaded.

I know this simple HTTP POST does not use much data/cpu/resources/energy.
But IMHO, it is still an unnecessary spam of 'unimportant' information going through the net, every 10 seconds, while much less frequent updates do the same job.
At some point it might come to be used by thousands of people, all spamming the server every 10 seconds.

I am not asking for a change here to be rushed, but maybe if the creator agrees and some members of github, who stumble on this discussion thread, can help decide. Let's see.

**Comment by Miskamyasa** [comment]
Thank you so much for your interest in the plugin and for your detailed comment. 10 seconds is the standard timeout for plugins developed by the project authors themselves. Here are some links to repositories as examples:
[Atom](https://gitlab.com/code-stats/code-stats-atom/-/blob/master/lib/code-stats-atom.js?ref_type=heads#L13)
[Sublime Text](https://gitlab.com/code-stats/code-stats-sublime/-/blob/master/CodeStats.py?ref_type=heads#L11)
[Browser](https://gitlab.com/code-stats/code-stats-browser/-/blob/master/js/content.js?ref_type=heads#L28)

**Reply by Miskamyasa** [reply]
As far as I understand, a larger timeout can disrupt the code flow counter

**Comment by Miskamyasa** [comment]
Regarding the placement of the plugin on the official website, I have conveyed the information to the service developers, and they promised to add it. Unfortunately, they lack the resources to do it promptly. On my end, I lack the skills to work with Elixir, the language on which the project's website is built. If you have the opportunity, I would be very grateful for your help.
